File: src/main/java/example/Role.java
package example;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.DiscriminatorColumn;
import javax.persistence.DiscriminatorType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.Table;

import org.hibernate.annotations.ForceDiscriminator;
import org.hibernate.annotations.Immutable;

@Entity
@Immutable
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "role_nam", discriminatorType = DiscriminatorType.STRING)
@ForceDiscriminator
@Table(name="z_roles")
public abstract class Role implements Comparable<Role>, Serializable {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = -6318360522731138654L;
	
	@Id
	@GeneratedValue
	@Column(name="role_id", insertable=false, updatable=false)
	private Long id;
	
	@Column(name="role_nam", insertable=false, updatable=false)
	private String name;

	public int compareTo(Role o) {
		return name.compareTo(o.getName());
	}

	
	@Override
	public int hashCode()
	{
		return name.hashCode();
	}
	
	@Override
	public boolean equals(Object other)
	{
		if (!(other instanceof Role)) return false;
		Role r = (Role) other;
		return (name.equals(r.getName()));
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getId() {
		return id;
	}

	public void setName(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

}

File: src/main/scala/example/Role.scala
package example

import java.io.Serializable
import javax.persistence._

@Entity 
@Table { val name="z_roles" }
class Role extends Serializable with Comparable[Role]
{

  @Id
  @GeneratedValue
  @Column { val name="role_id" }
  var id: java.lang.Integer = _

  @Column { val name="role_nam" }
  var name: String = _

  def compareTo(other: Role) = name.compareTo(other.name)

  override def equals(oth: Any) = { name == oth.asInstanceOf[Role].name }

  override def hashCode = name.hashCode
  
}

